client/state/data-layer/wpcom/login-2fa/index.js

```diff
@@ -39,15 +39,19 @@ const doAppPushRequest = ( store ) => {
 		.accept( 'application/json' )
 		.send( {
 			user_id: getTwoFactorUserId( store.getState() ),
-			two_step_nonce: getTwoFactorAuthNonce( store.getState() ),
+			two_step_nonce: getTwoFactorAuthNonce( store.getState(), 'push' ),
 			remember_me: getTwoFactorRememberMe( store.getState() ),
 			two_step_push_token: getTwoFactorPushToken( store.getState() ),
 			client_id: config( 'wpcom_signup_id' ),
 			client_secret: config( 'wpcom_signup_key' ),
 		} ).then( () => {
 			store.dispatch( { type: TWO_FACTOR_AUTHENTICATION_PUSH_POLL_COMPLETED } );
 		} ).catch( error => {
-			store.dispatch( { type: TWO_FACTOR_AUTHENTICATION_PUSH_UPDATE_NONCE, twoStepNonce: error.response.body.data.two_step_nonce } );
+			store.dispatch( {
```

Comment:
@Tug: I've just gave it a try by going offline while polling but as far I can see `catch` is never called. It looks like some middleware handles this specific use case:

<img width="582" alt="screenshot" src="https://cloud.githubusercontent.com/assets/594356/26573101/bcf3c580-451c-11e7-9416-e27590c99973.png">